fw/t/fuzzer.c

```diff
@@ -353,36 +373,87 @@ gen_vector_move(TfwFuzzContext *ctx, int i)
 }
 
 static void
-addch(char **p, char *end, char ch)
+h2_write_int(unsigned long index, char **p, char *end)
 {
-	if (!ch)
-		return;
+	static const unsigned short max = 0x7F;
+	static const unsigned short mask = 0x0;
+
+	if (likely(index < max)) {
+		index |= mask;
+	}
+	else {
+		*(*p)++ = max | mask;
+		index -= max;
+		while (index > max) {
+			if(*p + 1 > end)
+				return;
+			*(*p)++ = (index & 0x7F) | 0x80;
+			index >>= 7;
+		}
+	}
 
 	if(*p + 1 > end)
 		return;
 
-	*(*p)++ = ch;
+	*(*p)++ = index;
+}
+
+static void
+addch(TfwFuzzContext *ctx, char **p, char *end, char ch)
+{
+	BUG_ON(!ch);
+
+	if(*p == end) {
+		ctx->is_buf_not_enough = true;
+	}
+	else
+	{
+		*(*p)++ = ch;
+	}
+}
+
+static void
+add_string_n(TfwFuzzContext *ctx, char **p, char *end, const char *str, int n)
+{
+	while (*p != end && n) {
+		*(*p)++ = *str++;
+		--n;
+	}
```

Comment:
why not memcpy()?

Reply:
Seemed so much elegant). I changed the code to use memcpy.